book/style_guide.md

```diff
@@ -0,0 +1,180 @@
+   2. **It's recommended to** put one space after comma `,` after block or closure parameter parameter.
+   3. **It's recommended to** put no space before and one space after pipe `|` symbol denoting block or closure parameter list end.
+2. block and closure bodies:
+   1. **It's recommended to** put one space before closing block or closure curly brace `}`.
+3. records:
+   1. **It's recommended to** put one space before first record key and after last record key value.
+   2. **It's recommended to** put one space after `:` after record key.
+   3. **It's recommended to** put one space after comma `,` after key value.
+4. lists:
+   1. **It's recommended to** put no spaces before first list value and after last list value.
+   2. **It's recommended to** put one space after comma `,` after list value.
+5. surrounding constructs:
+   1. **It's recommended to** put one space before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one space after closing square `]` or curly brace `}` if following symbol is not the same.
+   3. **It's recommended to** put no spaces between square `[]` or curly brackets `{}` with nothing between them.
+
+Correct:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el| $el.status == UP }
+[1 2 3 4] | reduce {|it, acc| $it + $acc }
+{ x: 1, y: 2 }
+[1 2] | zip [3 4]
+[]
+```
+
+Incorrect:
+
+```nushell
+[[status]; [UP] [UP]] | all { |el| $el.status == UP } # two many spaces before "|el|": 2 instead of 0
+[1 2 3 4] | reduce {|it , acc | $it + $acc } # two many spaces after "it": 1 instead of 0
+                                             # two many spaces before "| $it + $acc": 1 instead of 0
+{x: 1, y : 2 } # two few spaces before "x: 1": 0 instead of 1
+               # two many spaces before ": 2": 1 instead of 0
+[1 2] | zip  [3 4] # two many spaces before "[3 4]": 2 instead of 1
+[ ] # two many spaces before "]": 1 instead of 0
+```
+
+#### Multi-line format
+
+Multi-line format is a format for writing all commands in several lines. It inherits all rules from one-line format
+and modifies them slightly.
+
+1. general:
+   1. **It's recommended to omit** trailing spaces.
+2. block and closure bodies:
+   1. **It's recommended to** put each body pipeline on a separate line.
+   2. **It's recommended to** put each body pipeline on a separate line.
+3. records:
+   1. **It's recommended to** put each record key-value pair on separate line.
+4. lists:
+   1. **It's recommended to** put each list item on separate line.
+5. surrounding constructs:
+   1. **It's recommended to** put one `\n` before opening square `[` or curly brace `{` if preceding symbol is not the same.
+   2. **It's recommended to** put one `\n` after closing square `]` or curly brace `}` if following symbol is not the same.
+
+Correct:
+
+```nushell
+[[status]; [UP] [UP]] | all {|el|
+    $el.status == UP
+}
+
+[
+    1
+    2
+    3
+    4
+] | reduce {|it, acc|
+    $it + $acc
+}
```

Comment:
personally, i wouldn't use this format because it's way too spaced out.
I'd go with this
```
[ 1 2 3 4 ] | reduce { |it acc|
    $it + $acc
}
```
or this
```
[ 1 2 3 4 ] | reduce { |it acc| $it + $acc }
```

Reply:
Just updated examples. I've explained in **Spreading long lines** that:

```nushell
[1, 2, 3, 4] | reduce {|it, acc|
    $it + $acc
}
```

is preferred in scripts.

Reply:
Maybe I'm just confused then because personally, I wouldn't recommend this format.
```
[
  {
    name: "Teresa",
    age: 24
  },
  {
    name: "Thomas",
    age: 26
  }
]
```
I'd want to see something maybe more like this. It kind of depends on what the input data looks like and how long it is. I really just don't want to have lines with 1 character on them.
```
[
  { name: "Teresa" age: 24 }
  { name: "Thomas" age: 26 }
]
```

Reply:
I've change rule to:

> - **It's recommended to** default to short format in scripts for lists and records unless they more than 80 characters long
  or contain nested lists or records.

In other words, it tells that:

```
[
  { name: "Teresa" age: 24 }
  { name: "Thomas" age: 26 }
]
```

is preferred.

Reply:
agree with @fdncred it's super spaced-out :open_mouth: 

i'd vote for merging these two sections into one and just using the compromise in "*Spreading long lines*"

i'd never write or want to read
```
[
    1
    2
    3
    4
] | reduce {|it, acc|
    $it + $acc
}
```
neither in the REPL or in scripts :scream: 

```
[1 2 3 4] | reduce {|it, acc|
    $it + $acc
}
```
is sexy on the other hand :smirk:

Reply:
so there's just this thread :smirk: 

what do you think @fdncred and @EmilySeville7cfg ? :yum:

Reply:
Just removed section about spreading lines, while explaining what format when should be used in a dedicated format sections.